router/core/flushwriter.go

```diff
@@ -73,14 +100,48 @@ func (f *HttpFlushWriter) Flush() (err error) {
 		f.cancel()
 		return
 	}
-	_, err = f.writer.Write([]byte("\n\n"))
+	separation := "\n\n"
+	if f.multipart {
+		separation = "\n"
+	}
+	_, err = f.writer.Write([]byte(separation))
 	if err != nil {
 		return err
 	}
 	f.flusher.Flush()
 	return nil
 }
 
+func (f *HttpFlushWriter) StartHeartbeat() {
+	if f.multipart {
+		f.ticker = time.NewTicker(multipartHeartbeatInterval * time.Second)
+		go func() {
+			for {
```

Comment:
It's more idiomatic when we release resources here. We can stop the ticker with `defer ticker.Stop()`